### ISSUE TITLE
Timeouts with geturls.get_urls_from_imap

### ISSUE THREAD

**Issue by MrTyton** [JimmXinu/FanFicFare#165]
imaplib does not by default include a timeout on the request. This can cause the request (and thus the program) to hang when calling the server, line [190](https://github.com/JimmXinu/FanFicFare/blob/master/fanficfare/geturls.py#L190), since it never gives up on the connection and tries again (or just abandons completely).

I'm not sure how you handle timeouts in anything else (for the writers and such) but the simplest way is

```
import socket
socket.setdefaulttimeout(30)
```

at some point before the line is called.

**Comment by JimmXinu** [comment]
`setdefaulttimeout` would effect _all_ new connections.  That's a problem for the Calibre plugin version; Calibre has lots of other possible connections.

Plus, it doesn't sound like it's that simple.  [This stackover answer](http://stackoverflow.com/questions/24422724/does-pythons-imaplib-let-you-set-a-timeout) lead me to [this python API doc](https://docs.python.org/2/library/socket.html#socket.socket.makefile) for `socket.makefile`: 
> The socket must be in blocking mode (it can not have a timeout). 

What other cases might there be where blocking is necessary?  I presume the python `imaplib` doesn't include a timeout option for good reasons, such as that.

That said, if you'd like to contribute code that adds something like [the stackover answer](http://stackoverflow.com/questions/24422724/does-pythons-imaplib-let-you-set-a-timeout), I'll accept it.  But only as a default-off optional feature.

**Comment by MrTyton** [comment]
I've run into it before. From my understanding ([here ](http://stackoverflow.com/questions/21602924/what-is-the-proper-way-to-set-a-global-socket-timeout-in-python)and in the linked in the answer), setdefaulttimeout only affects the sockets created in the python file that it's currently in, not any others, so the timeout clause would only apply to the functions that are within geturls.py.

From modifying geturls.py on my machine, sticking those two lines in the beginning, it didn't affect the functionality when I ran it through FFF CLI (and there definitely has been some times that it has timed out, which was the reason that I started looking into this to begin with). The reason that the socket has to be in blocking mode is because it treats it like a file pointer, I think.

I can modify it to be specific for this with optionality included from the config file, if you want, that isn't a problem. I'll get it done later today.
